### PR TITLE
Compat: Test texture+sampler limits are enforced

### PR DESCRIPTION
**Pull request by greggman**
See: https://github.com/gpuweb/gpuweb/pull/4989

<hr>

**Requirements for PR author:**

- [X] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [X] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [X] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)
- [X] Test have be tested with compatibility mode validation enabled and behave as expected. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

File: src/webgpu/compat/api/validation/pipeline_creation.spec.ts [hunk 1]
```diff
@@ -144,3 +144,206 @@ g.test('depth_textures')
...
+          { pass: true,  numCombos: 'max',   numNonSampled: '1',   numExternal: 0, useSame: false },
+          { pass: false, numCombos: 'max+1', numNonSampled: '1',   numExternal: 0, useSame: false },
+          { pass: true,  numCombos: '1',     numNonSampled: 'max', numExternal: 0, useSame: false },
+          { pass: false, numCombos: '2',     numNonSampled: 'max', numExternal: 0, useSame: false },
+          { pass: true,  numCombos: 'max-4', numNonSampled: '0',   numExternal: 1, useSame: false },
+          { pass: false, numCombos: 'max-3', numNonSampled: '0',   numExternal: 1, useSame: false },
+          { pass: true,  numCombos: 'max-8', numNonSampled: '0',   numExternal: 2, useSame: false },
+          { pass: false, numCombos: 'max-7', numNonSampled: '0',   numExternal: 2, useSame: false },
+          { pass: true,  numCombos: 'max-7', numNonSampled: '0',   numExternal: 2, useSame: true },
+          { pass: false, numCombos: 'max-6', numNonSampled: '0',   numExternal: 2, useSame: true },
```
Comment (shrekshao) [inline src/webgpu/compat/api/validation/pipeline_creation.spec.ts:195]: nit: it'd be good to have comments or language in description explaining each external texture/sampler takes 3 slot (YUV) IIUC

File: src/webgpu/compat/api/validation/pipeline_creation.spec.ts [hunk 2]
```diff
@@ -144,3 +144,206 @@ g.test('depth_textures')
...
+  ${usages[1].join('\n')}
+  return c;
+}
+
+@vertex fn vs() -> @builtin(position) vec4f {
+  _ = ${stages.includes('vertex') ? 'usage0()' : 'vec4f(0)'};
+  return vec4f(0);
+}
+
+@group(2) @binding(0) var tt: texture_2d<f32>;
```
Comment (shrekshao) [inline src/webgpu/compat/api/validation/pipeline_creation.spec.ts:316]: Q: why do have this?

Comment (greggman) [reply src/webgpu/compat/api/validation/pipeline_creation.spec.ts:316]: missed that. It was a test. Removed.

File: src/webgpu/compat/api/validation/pipeline_creation.spec.ts (hunk 1 above)
Comment (greggman) [reply src/webgpu/compat/api/validation/pipeline_creation.spec.ts:195]: I don't mind adding comments but is this a pattern from other tests? Can you give an example of what you want to see? I can copy and paste the spec but that seems wrong given the spec might change and I don't see the spec being copy and pasted in other tests. To see what's being tested shouldn't you just reference the spec? I started to write a comment per line but after writing 5 I realized I was just repeating the data in the params.

Comment (shrekshao) [reply src/webgpu/compat/api/validation/pipeline_creation.spec.ts:209]: Okay then let's just leave it as is. Future contributors can just rely on spec.
(Context: I was a bit confused on external texture takes 3 on first glance so I made this comment.)

**Review by shrekshao — APPROVED** [review]
LGTM

**Pull request merged**